src/mscorlib/shared/System/Diagnostics/Debug.cs

```diff
@@ -91,18 +92,35 @@ public static void Assert(bool condition, string message, string detailMessage)
             if (!condition)
             {
                 string stackTrace;
-
                 try
                 {
-                    stackTrace = Internal.Runtime.Augments.EnvironmentAugments.StackTrace;
+                    stackTrace = PopStackTrace(Internal.Runtime.Augments.EnvironmentAugments.StackTrace);
                 }
                 catch
                 {
                     stackTrace = "";
                 }
+                WriteLine(FormatAssert(stackTrace, message, detailMessage));
+                s_ShowDialog(stackTrace, message, detailMessage, (uint)ErrorSource.Assert);
+            }
+        }
 
+        [System.Diagnostics.Conditional("DEBUG")]
+        public static void Assume(bool condition, string message, string detailMessage)
```

Comment:
Lets make this internal. I don't think there was any intent to add new public APIs and if we want to add it later we can always change it then.

Reply:
The naming here appears to favor 'Assume' but there are a number of other kinds of contract method such as Invariant or Requires. I'd suggest doing ContractFail(string message, string detailMessage). I would correspondingly change the error text that gets printed to "Contract Failed" unless @jkotas was particular that it really be "Assumption failed"?

Reply:
> Assumption failed

It is what it prints on .NET Framework. I am not particular about the name. ContrailFail sounds fine to me. BTW: A related method is called `OnContractFailure` in CoreRT/ProjectN.

Reply:
Ah if that is what .NET framework prints then I'd gladly make the behavior match. I thought we had inherited this bad behavior from .Net Framework however. Do you happen to know if something specifically regressed the .Net Core behavior? If you don't know offhand no worries, Sung and I can compare them.

Thanks Jan!

Reply:
The specific change that touched the `Assumption failed` message was my #14868. Contracts used its own path to display error message with its own set of issues. When I have switched them to use the same path as asserts, I have found that asserts are broken in multiple different ways so I just left TODO there.

Other changes that contributed to this:
- Original refactoring of `System.Diagnostic.Debug` namespace for CoreCLR and ProjectN
- Assert was changed to now show message box by default in .NET Core (message boxes do not make sense in cloud environments)
- FailFast was changed to print the stacktrace in .NET Core (it did nothing in .NET Framework). This change caused stacktrace to be printed twice for Assert.